src/Endpoint/Abstracts/AbstractSugarBeanEndpoint.php

```diff
@@ -367,6 +368,34 @@ public function relate(string $linkName, string $related_id): AbstractSugarBeanE
         return $this->link($linkName, $related_id);
     }
 
+    /**
+     * Fetch Audits as a Collection on Current Record
+     * @return AbstractSugarBeanCollectionEndpoint
+     */
+    public function auditLog(): AbstractSugarBeanCollectionEndpoint
+    {
+        $auditCollection = new ModuleAudit(['module' => $this->_beanName, 'id' => $this->get('id')]);
+
+        $versionUpdated = false;
+        $client = $this->getClient();
+        if ($client){
+            $auditCollection->setClient($client);
+            $originalClientVersion = $client->getVersion();
+            // check if client version is older than 11_11
+            if (version_compare($originalClientVersion, "11_11", "<")) {
+                $client->setVersion("11_12");
+                $versionUpdated = true;
+            }
+        } else {
+            $auditCollection->setBaseUrl($this->getBaseUrl());
+        }
+
```

Comment:
you should call $auditCollection->fetch() here. Otherwise the changing of version and setting it back, doesn't do anything